app/tests/integration/models/validation_dataclass.py

```diff
@@ -0,0 +1,36 @@
+from typing import List, Any, Optional, Dict
+from pydantic import BaseModel, Field
+
+
+class FilterObjectDataClass(BaseModel):
+    type: Optional[str]
+    channels: Optional[List]
+    versions: Optional[List]
+
+
+class PreferencesDataClass(BaseModel):
+    preferenceBranchType: str
+    preferenceType: str
+    preferenceValue: Any
+
+
+class BranchDataClass(BaseModel):
+    slug: str
+    ratio: Optional[str]
+    addon_url: str = Field(default=None)
+    preferences: Dict[Optional[str], PreferencesDataClass] = Field(default=None)
```

Comment:
This doesn't look correct. The type annotations suggests the preferences `Dict` is always present, but it may contain either `str` or `None` keys.

Do you mean?

```suggestion
    preferences: Optional[Dict[str, PreferencesDataClass]] = Field(default=None)
```